docs/javascript/01-getting-started.md

```diff
@@ -17,14 +17,21 @@ The most basic usage is to request an instance of a class definition. TypeDI wil
 been created before and return the cached version or it will create a new instance, cache and return it.
 
 ```js
+import 'reflect-metadata';
 import { Container } from 'typedi';
 
+// require('reflect-metadata');
+// const { Container }= require('typedi');
+
 class ExampleClass {
   print() {
     console.log('I am alive!');
   }
 }
 
+/** Register this class to the TypeDI container */
+Container.set({ id: ExampleClass, type: ExampleClass});
```

Comment:
I have not tested this but this seems a bit odd to me. Shouldn't it be 
```typescript
Container.set({id: ExampleClass, value: new ExampleClass});
```
?

Reply:
@attilaorosz Yes this can work, but using "value" instead of "type" can cause problems when you include it with multiple classes that depend on one another. Although using value will be fine, I don't want to confuse our doc readers when they hit certain problems with it's limitation. I will provide an example:

First, the docs state that you can use either value, type, or factory. https://github.com/typestack/typedi/blob/develop/src/interfaces/service-options.interface.ts#L3

The problems that will occur with using value instead of type is that the _order in which you register the classes will matter_

Example of using registering classes, in which one depends on the other but is registered in the wrong order.


```js
require('reflect-metadata');
const { Container } = require('typedi');

class EmailService {
    sendEmail(email) {
        console.log("Sending e-mail to the following address.", { email });
    }
}

class UserService {
    constructor(container) {
        this.emailService = container.get(EmailService);
    }
    signUp(email) {
        console.log("Signing user up!");
        this.emailService.sendEmail(email);
    }
}

/**
 * Register these classes to the TypeDI container.
 * 
 * This will cause a TypeError of "Cannot read property 'get' of undefined".
 * Because since we initialized the class here, the container doesn't get passed to the constructor
 */
Container.set({ id: UserService, value: new UserService });
Container.set({ id: EmailService, value: new EmailService });

const classInstance = Container.get(UserService);

classInstance.signUp("happyzombies@gmail.com");
```

So the above does not work because registering the class by passing the value instantiates the class instantly when we aren't necessarily passing container to it.

For the next example, let's say we do pass container (or just call Container directly from the constructor). Well if we did not register the EmailService first, we will get another error. Which this becomes annoying because me registering these services, I then **have** to remember what order to register them in (which kind of defeats the whole purpose of this module).

```js
require('reflect-metadata');
const { Container } = require('typedi');

class EmailService {
    sendEmail(email) {
        console.log("Sending e-mail to the following address.", { email });
    }
}

class UserService {
    constructor() {
        this.emailService = Container.get(EmailService);
    }
    signUp(email) {
        console.log("Signing user up!");
        this.emailService.sendEmail(email);
    }
}

/**
 * Register these classes to the TypeDI container.
 * 
 * This will cause a ServiceNotFoundError. Because we need to initialize the EmailService class first.
 */
Container.set({ id: UserService, value: new UserService });
Container.set({ id: EmailService, value: new EmailService });

const classInstance = Container.get(UserService);

classInstance.signUp("happyzombies@gmail.com");
```

To not cause the user confusion when reading the docs, I set this to register with "type" because the classes we register _will not be initialized until the first time we call .get()_. Which is great because the order in which we need to register the class does not matter.

Working example with type.

```js
require('reflect-metadata');
const { Container } = require('typedi');

class EmailService {
    sendEmail(email) {
        console.log("Sending e-mail to the following address.", { email });
    }
}

class UserService {
    constructor() {
        this.emailService = Container.get(EmailService);
    }
    signUp(email) {
        console.log("Signing user up!");
        this.emailService.sendEmail(email);
    }
}

/**
 * Register these classes to the TypeDI container using Type. UserService is registered first but NOT initialized until the first .get() is called. No errors and it works.
 */
Container.set({ id: UserService, type: UserService });
Container.set({ id: EmailService, type: EmailService });

const classInstance = Container.get(UserService);

classInstance.signUp("happyzombies@gmail.com");
```

Reply:
That's some deep investigation that I appreciate :). I see the problem and I agree with your solution, but probably the docs should reflect this.
If you could include these limitations and the fact the container gets injected into the constructor would be really helpful to users!

Reply:
Okay I'll include a little bit about these limitations, these are the "get started quickly" docs, so I feel like it should be straight forward to get working. 

I'll make another PR to update the advanced doc sections to go more into this. 👍

Reply:
@attilaorosz checkout what I added. I believe more specific details can be added later on. But this is just a get started quick guide to get people using the module quickly. Lmk and thanks